django/contrib/admin/options.py

```diff
@@ -1214,12 +1215,31 @@ def construct_search(field_name):
             for bit in smart_split(search_term):
                 if bit.startswith(('"', "'")) and bit[0] == bit[-1]:
                     bit = unescape_string_literal(bit)
-                or_queries = models.Q.create(
-                    [(orm_lookup, bit) for orm_lookup in orm_lookups],
-                    connector=models.Q.OR,
-                )
-                term_queries.append(or_queries)
-            queryset = queryset.filter(models.Q.create(term_queries))
+
+                field_queries = []
+                for orm_lookup in orm_lookups:
+                    try:
+                        # Attempt to create a query for each field
+                        field_query = models.Q(**{orm_lookup: bit})
+                        queryset.filter(
+                            field_query
+                        ).exists()  # Test if the query is valid
```

Comment:
This is better but this adds extra queries
I think we might be able to annotate values which are cast to strings and search these when the lookup is "exact" and it is not a Char or TextField

Reply:
Ah, good 👀.
Will fix soon.